File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/database/witnessing/WitnessEntity.java
package com.github.dedis.popstellar.repository.database.witnessing;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;

import com.github.dedis.popstellar.model.objects.security.PublicKey;

@Entity(
    tableName = "witnesses",
    primaryKeys = {"lao_id", "witness"})
public class WitnessEntity {

  @ColumnInfo(name = "lao_id")
  @NonNull
  private final String laoId;

  @ColumnInfo(name = "witness")
  @NonNull
  private final PublicKey witness;

  public WitnessEntity(@NonNull String laoId, @NonNull PublicKey witness) {
    this.laoId = laoId;
    this.witness = witness;
  }

  @NonNull
  public String getLaoId() {
    return laoId;
  }

  @NonNull
  public PublicKey getWitness() {
    return witness;
  }
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/database/witnessing/WitnessEntity.kt
package com.github.dedis.popstellar.repository.database.witnessing

import androidx.room.ColumnInfo
import androidx.room.Entity
import com.github.dedis.popstellar.model.objects.security.PublicKey

@Entity(tableName = "witnesses", primaryKeys = ["lao_id", "witness"])
class WitnessEntity(
    @field:ColumnInfo(name = "lao_id") val laoId: String,
    @field:ColumnInfo(name = "witness") val witness: PublicKey
)
